sdk/mobile-center-analytics/src/main/java/com/microsoft/azure/mobile/analytics/Analytics.java

```diff
@@ -277,36 +288,36 @@ private static boolean validateName(String name, String logType) {
         if (properties == null)
             return null;
         String message;
-        final int maxPropertiesCount = 5;
-        final int maxPropertyItemLength = 64;
         Map<String, String> result = new HashMap<>();
         for (Map.Entry<String, String> property : properties.entrySet()) {
-            if (result.size() >= maxPropertiesCount) {
-                message = String.format("%s '%s' : properties cannot contain more than %s items. Skipping other properties.", logType, logName, maxPropertiesCount);
+            String key = property.getKey();
+            String value = property.getValue();
+            if (result.size() >= MAX_PROPERTY_COUNT) {
+                message = String.format("%s '%s' : properties cannot contain more than %s items. Skipping other properties.", logType, logName, MAX_PROPERTY_COUNT);
                 MobileCenterLog.warn(Analytics.LOG_TAG, message);
                 break;
             }
-            if (property.getKey() == null || property.getKey().isEmpty()) {
+            if (key == null || key.isEmpty()) {
                 message = String.format("%s '%s' : a property key cannot be null or empty. Property will be skipped.", logType, logName);
                 MobileCenterLog.warn(Analytics.LOG_TAG, message);
                 continue;
             }
-            if (property.getKey().length() > maxPropertyItemLength) {
-                message = String.format("%s '%s' : property '%s' : property key length cannot be longer than %s characters. Property '%s' will be skipped.", logType, logName, property.getKey(), maxPropertyItemLength, property.getKey());
+            if (value == null) {
+                message = String.format("%s '%s' : property '%s' : property value cannot be null. Property '%s' will be skipped.", logType, logName, key, key);
                 MobileCenterLog.warn(Analytics.LOG_TAG, message);
                 continue;
             }
-            if (property.getValue() == null) {
-                message = String.format("%s '%s' : property '%s' : property value cannot be null. Property '%s' will be skipped.", logType, logName, property.getKey(), property.getKey());
+            if (key.length() > MAX_PROPERTY_ITEM_LENGTH) {
+                message = String.format("%s '%s' : property '%s' : property key length cannot be longer than %s characters. Property key '%s' will be truncated.", logType, logName, key, MAX_PROPERTY_ITEM_LENGTH, key);
```

Comment:
It was not in original tfs but I confirmed with @elamalani. But this comment you reply too is a different comment about message to print for event key and value (to align with .net)